### PR TITLE
Update string_iterator.c3 to include extra convenience methods

### PR DESCRIPTION
**Pull request by ellipse12**
Added peek: returns the next character without incrementing current
Added has_next: checks if the iterator has another element
Added get: gets the current element (the same one that was returned with the previous call to next).
Most of this is to make it easier to keep unicode support with common operations.

### CODE REVIEW

**Comment by ellipse12** [comment]
Oops! the get function has a bug it should be:
```c3
fn Char32! StringIterator.get(&self)
{
    usz len = self.utf8.len;
    usz current = self.current;
    usz read = (len - current < 4 ? len - current : 4);
    Char32 res = conv::utf8_to_char32(&self.utf8[current - read > 0 : current - read ? 0], &read)!;
    return res;
}
```
(changed it so that current is clamped to 0 and also gets the correct element)

**Comment by lerno** [comment]
Just update it and the pull request should update. I always squash pull requests in the end, so it doesn't matter if there are multiple commits.

**Comment by ellipse12** [comment]
Ok good to know! Just fixed it.

**Comment by lerno** [comment]
Looks like there's still an error?

**Comment by ellipse12** [comment]
Thank you for your patience. :)

**Comment by lerno** [comment]
Oh, I noticed there were no tests, can you add some tests? You can look into /test/unit/stdlib for examples. Just something to check what happens if each is called in the beginning, in the end and at the end where there is an invalid UTF8 sequence.

**Comment by ellipse12** [comment]
Ok ill do that.

**Comment by ellipse12** [comment]
Done! Sorry for all of the hassle this is my first PR so thank you for being patient with me. Also just wondering how do you build the Debian binaries? The default Debian package repo does not have llvm-17. (I ended up just using a virtual machine to test with)

File: lib/std/core/string_iterator.c3 [hunk 1]
```diff
@@ -1,5 +1,5 @@
 module std::core::string::iterator;
-
+import std::io;
```
Comment (lerno) [inline lib/std/core/string_iterator.c3:2]: Is this needed?

**Comment by lerno** [comment]
I grab the llvm binaries from the LLVM org directly. If you look at this:
https://github.com/c3lang/c3c/blob/master/.github/workflows/main.yml

You can scroll down and see exactly the commands that are run to build the compiler for the different targets. You should be able to figure it out even if you don't know how Github defines things, because the actual commands are just direct commands send to the command line for each machine.

**Comment by lerno** [comment]
Also, the first PR and getting the hang of what's needed is always a bit of a learning experience. Don't worry about it.

File: lib/std/core/string_iterator.c3 (hunk 1 above)
Comment (lerno) [reply lib/std/core/string_iterator.c3:2]: If not then we should remove it.

Comment (ellipse12) [reply lib/std/core/string_iterator.c3:2]: Yeah no it's not I just forgot to remove it after testing.

**Comment by ellipse12** [comment]
Thanks for your understanding!

**Comment by lerno** [comment]
Thank you!

**Pull request merged**